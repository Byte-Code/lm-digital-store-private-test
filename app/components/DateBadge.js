import React, { Component } from 'react';
import moment from 'moment';
import styled from 'styled-components';

moment.locale('it');

const Time = styled.div`
  font-size: 48px;
  color: #fff;
  font-family: LeroyMerlinSans Bold;
  text-transform: uppercase;
`;

export default class DateBadge extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentTime: '',
      intID: ''
    };
  }

  componentWillMount() {
    const intID = this.updateTime();
    this.setState({ intID });
  }

  shouldComponentUpdate(nextProps, nextState) {
    return nextState.currentTime !== this.state.currentTime;
  }

  componentWillUnmount() {
    const { intID } = this.state;
    clearInterval(intID);
  }

  updateTime() {
    return setInterval(() => {
      const currentTime = moment().format('ddd HH:mm');
      this.setState({ currentTime });
    }, 1000);
  }

  render() {
    const { currentTime } = this.state;
    return (
      <Time>
        {currentTime}
      </Time>
    );
  }
}
