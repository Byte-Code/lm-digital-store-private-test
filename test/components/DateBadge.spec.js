import React from 'react';
import { mount } from 'enzyme';
import moment from 'moment';

import DateBadge from '../../app/components/DateBadge';

moment.locale('it');

describe('DateBadge', () => {
  const result = mount(<DateBadge />);

  it('should update state with the intervalID before mounting', () => {
    expect(result.state('intID')).toEqual(0);
  });

  it('should update state with the currentTime everySecond', () => {
    setTimeout(() => {
      const currentTime = moment().format('ddd HH:mm');
      expect(result.state('currentTime')).toEqual(currentTime);
    }, 1000);
  });

  it('should render properly', () => {
    expect(result).toMatchSnapshot();
  });
});
