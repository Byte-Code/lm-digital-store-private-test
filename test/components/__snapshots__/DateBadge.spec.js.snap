// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateBadge should render properly 1`] = `
<DateBadge>
  <glamorous(div)>
    <div
      className="css-a7u1rk"
    />
  </glamorous(div)>
</DateBadge>
`;
